packages/api-clients/open-api/bffApi.yml

```
@@ -26275,10 +26400,11 @@ components:
         title:
           type: string
         riskAnalysisForm:
-          $ref: "#/components/schemas/RiskAnalysisFormSeed"
           description: |
             Optional in the purpose model, but a purpose cannot exist without a risk analysis.
             There is no practical use in letting the user remove it, we don't make it nullable.
+          allOf:
+            - $ref: "#/components/schemas/RiskAnalysisFormSeed"
```

Is this change correct?
This `$ref` seemed correct,  while now we have `allOf:` with a single reference below...
Same for other occurrences of this similar change

---

Yes, this change is correct

The thing is, in OpenAPI 3.0 sibling properties next to $ref are simply ignored. So the old code with $ref and description side by side wasn't actually working - the description was just being silently discarded.

The allOf wrapper with a single $ref is the standard workaround to keep the description while still using the reference. It's documented here: https://swagger.io/docs/specification/v3_0/using-ref/#considerations

> "When using $ref, all sibling properties are ignored. To add a description or override properties, you need to wrap the $ref in allOf."

allOf doc: https://swagger.io/docs/specification/v3_0/data-models/oneof-anyof-allof-not/#allof

(This quirk was fixed in OpenAPI 3.1 where you can have siblings with $ref, but we're on 3.0.3 )

---

I think then we should probably turn off this warning, it makes the readability of the spec less obvious 🥲 
Let's revert these single `$ref` to `allOf` changes and keep the rule off if possible.